integration-tests/smoke/keeper_test.go

```diff
@@ -377,6 +383,72 @@ func getKeeperSuite(
+								"Expected consumer counter to be greater than 0, but got %d", counter.Int64())
+							log.Info().Int64("Upkeep counter", counter.Int64()).Msg("Upkeeps performed")
+						}
+					}, "1m", "1s").Should(Succeed())
+
+					newConsumers, newUpkeepIDs := actions.RegisterNewUpkeeps(contractDeployer, networks, linkToken,
+						registry, registrar, upkeepGasLimit, 1)
+
+					// We know that newConsumers has size 1, so we can just use the newly registered upkeep.
+					newUpkeep := newConsumers[0]
+					newUpkeepID := newUpkeepIDs[0]
+
+					// Test that the newly registered upkeep is also performing.
+					Eventually(func(g Gomega) {
+						counter, err := newUpkeep.Counter(context.Background())
+						g.Expect(err).ShouldNot(HaveOccurred(), "Calling newly deployed upkeep's counter shouldn't fail")
+						g.Expect(counter.Int64()).Should(BeNumerically(">", int64(0)),
+							"Expected newly registered upkeep's counter to be greater than 0, but got %d", counter.Int64())
+						log.Info().Int64("Upkeep counter", counter.Int64()).Msg("Upkeeps performed")
+					}, "1m", "1s").Should(Succeed())
+
+					// While we registered the new upkeep, the initial upkeeps that were created first should have
+					// kept performing and hence their counters should be larger than what they were before.
+					for i := 0; i < len(upkeepIDs); i++ {
+						currentCounter, err := consumers[i].Counter(context.Background())
+						Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's counter shouldn't fail")
+						log.Info().Int64("Upkeep counter", currentCounter.Int64()).Msg("Upkeep cancelled")
+
+						Expect(initialCounters[i].Int64() <= currentCounter.Int64()).To(BeTrue())
+					}
+
+					// Test that we can also cancel the newly registered upkeep.
+					err := registry.CancelUpkeep(newUpkeepID)
+					Expect(err).ShouldNot(HaveOccurred(), "Upkeep should get cancelled successfully")
+					err = networks.Default.WaitForEvents()
+					Expect(err).ShouldNot(HaveOccurred(), "Error encountered when waiting for "+
+						"newly registered upkeep to be cancelled")
+
+					// Obtain the amount of times the new upkeep has been executed so far
+					counterAfterCancellation, err := newUpkeep.Counter(context.Background())
+					Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's counter shouldn't fail")
+					log.Info().Int64("Upkeep counter", counterAfterCancellation.Int64()).Msg("Upkeep cancelled")
+
+					// Make sure the counter stays constant because we cancelled the newly registered upkeep.
+					Consistently(func(g Gomega) {
+						latestCounter, err := newUpkeep.Counter(context.Background())
+						g.Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's counter shouldn't fail")
+						g.Expect(latestCounter.Int64()).Should(Equal(counterAfterCancellation.Int64()),
+							"Expected consumer counter to remain constant at %d, but got %d",
+							counterAfterCancellation.Int64(), latestCounter.Int64())
+					}, "1m", "1s").Should(Succeed())
```

Comment:
imo, we already test the cancellation behaviour in smoke test so we can remove this, upto you

Reply:
That's true, I just thought that this upkeep is registered separately to the other ones so wouldn't harm to check again, but yes it's not necessary.

Reply:
I would lean towards removing if we don't see value, o/w we can go down a rabbit hole of checking everything in every test